File: src/main/java/hdzi/editstarters/ui/GradleButtonAction.java
package hdzi.editstarters.ui;

import com.intellij.openapi.actionSystem.DataContext;
import hdzi.editstarters.buildsystem.BuildSystem;
import hdzi.editstarters.buildsystem.gradle.GradleBuildSystem;
import org.jetbrains.plugins.gradle.util.GradleConstants;

/**
 * Created by taojinhou on 2019/1/14.
 */
public class GradleButtonAction extends EditStartersButtonAction {
    @Override
    protected boolean isMatchFile(String name) {
        return GradleConstants.DEFAULT_SCRIPT_NAME.equals(name)
                || GradleConstants.KOTLIN_DSL_SCRIPT_NAME.equals(name);
    }

    @Override
    protected BuildSystem newBuildSystem(DataContext dataContext) {
        return GradleBuildSystem.of(dataContext);
    }
}

File: src/main/kotlin/hdzi/editstarters/ui/GradleButtonAction.kt
package hdzi.editstarters.ui

import com.intellij.openapi.actionSystem.DataContext
import hdzi.editstarters.buildsystem.gradle.GradleBuildSystem
import hdzi.editstarters.ui.dialog.EditStartersButtonAction
import org.jetbrains.plugins.gradle.util.GradleConstants

/**
 * Created by taojinhou on 2019/1/14.
 */
class GradleButtonAction : EditStartersButtonAction() {
    override fun String.isMatchFile(): Boolean =
        this == GradleConstants.DEFAULT_SCRIPT_NAME || this == GradleConstants.KOTLIN_DSL_SCRIPT_NAME

    override fun newBuildSystem(dataContext: DataContext) = GradleBuildSystem(dataContext)
}